examples/end-to-end-examples/finetuning/finetune_small_dataset.py

```diff
@@ -0,0 +1,262 @@
+        integrations=integrations,
+        command=command,
+    )
+    run = create_run(run_config)
+    print(f'Created convert to hugging face run with name: {run.name}.\n Tail the logs with `mcli logs --follow {run.name}`')
+    return run
+
+
+def create_deployment(
+    cluster_name: str,
+    num_replicas: int,
+    hugging_face_path: str,
+    model_name: str,
+) -> InferenceDeployment:
+    compute_config = {
+        'cluster': cluster_name,
+        'instance': 'oci.vm.gpu.a10.1',
+        'gpu_type': 'a10',
+        'gpus': 1}
+    
+    integrations = [{
+        'integration_type': 'git_repo',
+        'git_repo': 'mosaicml/examples',
+        'git_branch': 'main',
+        'ssh_clone': False,
+    },
+    {
+        'integration_type': 'git_repo',
+        'git_repo': 'mosaicml/llm-foundry',
+        'git_commit': '2c92faa5ce31888214bdb582ac7f5756d0d3dacd',
+        'ssh_clone': False,
+    }]
+
+    command = """
+        export PYTHONPATH=$PYTHONPATH:/code/llm-foundry:/code/examples:/code
+        pip uninstall packaging -y
+        rm /usr/lib/python3/dist-packages/packaging-23.1.dist-info/REQUESTED
+        pip install composer[streaming,libcloud,oci]==0.14.1
+        pip install packaging==23.1
+    """
+
+    model = {
+        'downloader': 'examples.end-to-end-examples.sec_10k_qa.deployment_download_helper.download_and_convert',
+        'download_parameters': {
+            'remote_uri': hugging_face_path
+        },
+        'model_handler': 'examples.inference-deployments.mpt.mpt_ft_handler.MPTFTModelHandler', # Use the provided MPT handler
+        'model_parameters': {
+            'ft_lib_path': '/code/FasterTransformer/build/lib/libth_transformer.so',
+            'model_name_or_path': f'mosaicml/{model_name}',
+            'gpus': 1,
+        },
+        'backend': 'faster_transformers',
+    }
+
+    deployment_config = InferenceDeploymentConfig(
+        name=f'finetune-deployment-{model_name}',
+        compute=compute_config,
+        integrations=integrations,
+        replicas=num_replicas,
+        command=command,
+        model=model,
+    )
+    deployment = create_inference_deployment(deployment_config)
+    print(f'Created inference deployment with name: {deployment.name}, DNS: {deployment.public_dns}')
+    print(f'Check the status with: `mcli get deployments` and logs with `mcli get deployment logs --follow {deployment.name}`')
+    return deployment
+
+
+if __name__ == '__main__':
+    # Data conversion
+    convert_data(
+        tokenizer_name=f'mosaicml/{MODEL_NAME.value}',
+        output_folder='oci://nancy-test',
+        train_index_path='oci://nancy-test/10k_train.jsonl',
+        eval_index_path='oci://nancy-test/10k_validate.jsonl',
+        concat_tokens=0,
+        eos_text='</s>',
+        bos_text='<s>',
+        no_wrap=True,
+        max_workers=1,
+        compression='gzip'
+    )
```

Comment:
have not tested this part yet -- the train_index_path/eval_index_path are incorrect right now